### ISSUE TITLE
Loftr backbone

### ISSUE THREAD

**Issue by chenjiajie9811** [crockwell/far#8]
Hi there, 

thank you for your amazing work! I have some questions regarding the loftr backbone. Do you use the pretrained loftr model for getting the feature and correspondences or train it from scratch? The model is supervised purely using the gt pose, is it strong enough for getting good correspondences from loftr?

**Comment by crockwell** [comment]
Hello,

LoFTR is trained from scratch. It is trained in the standard fashion: with losses on correspondences, where correspondences are derived from ground truth pose and depth. The pose prediction network (FAR) is supervised using only pose. Note this loss backpropogates to LoFTR, but is not sufficient to train correspondences alone. 

Hope this helps!
Chris

**Comment by chenjiajie9811** [comment]
Thank you crockwell for your explaination. 

I have a follow up question, since the pose loss of the pose prediction network backpropogates to LoFTR, do you observe any improvement on the performance of the LoFTR module alone? (Or does the LoFTR benefits from the end-to-end training with the pose branch, producing more accurate correspondences?)

**Comment by crockwell** [comment]
I didn't see a meaningful benefit in correspondence prediction. Early experiments showed backprop through pose prediction network to LoFTR might help very marginally

**Comment by chenjiajie9811** [comment]
Another question on the translation scale. 

I looked through the code and it seems that the predict_translation_scale in default setting and the training scripts you provided is set to False; And when the solver type is prior_ransac, the translation_scale is set to None [here](https://github.com/crockwell/far/blob/753517e99b032dce67d43deb68c0b5d3f07cd60b/mp3d_loftr/src/lightning/lightning_loftr.py#L156), so I am wondering whether the translation scale is different when you perform the linear combination of transformer regressed RT with the solver computed RT?

**Comment by crockwell** [comment]
Good question, I can see how that could be a bit confusing. That variable is for a separate head that could predict translation scale. For the experiments in the publicly released code, we use the network as-is, which predicts translation scale implicitly by predicting a translation vector which includes scale. So you can ignore the `predict_translation_scale` variable.

Check [this function](https://github.com/crockwell/far/blob/753517e99b032dce67d43deb68c0b5d3f07cd60b/mp3d_loftr/src/loftr/loftr_module/transformer.py#L350) for how we predict T vector.

**Comment by crockwell** [comment]
Closing for inactivity - please feel free to open again if you have further questions.
